lib/src/op_walk.rs

```diff
@@ -229,22 +229,28 @@ impl PartialOrd for OperationByEndTime {
 }
 
 /// Walks `head_ops` and their ancestors in reverse topological order.
-pub fn walk_ancestors(head_ops: &[Operation]) -> impl Iterator<Item = OpStoreResult<Operation>> {
+pub fn walk_ancestors(head_ops: &[Operation], reversed: bool) -> impl Iterator<Item = OpStoreResult<Operation>> {
     // Emit the latest head first to stabilize the order.
     let mut head_ops = head_ops
         .iter()
         .cloned()
         .map(OperationByEndTime)
         .collect_vec();
     head_ops.sort_unstable_by(|op1, op2| op1.cmp(op2).reverse());
-    // Lazily load operations based on timestamp-based heuristic. This works so long
-    // as the operation history is mostly linear.
-    dag_walk::topo_order_reverse_lazy_ok(
-        head_ops.into_iter().map(Ok),
-        |OperationByEndTime(op)| op.id().clone(),
-        |OperationByEndTime(op)| op.parents().map_ok(OperationByEndTime).collect_vec(),
-    )
-    .map_ok(|OperationByEndTime(op)| op)
+    let walk = if reversed {
+        dag_walk::topo_order_forward_ok(
+            head_ops.into_iter().map(Ok),
+            |OperationByEndTime(op)| op.id().clone(),
+            |OperationByEndTime(op)| op.parents().map_ok(OperationByEndTime).collect_vec(),
+        )
+    } else {
+        dag_walk::topo_order_reverse_ok(
```

Comment:
Dropping lazy version is undesired. Maybe better to handle `--reversed` by caller in the same way as `cmd_log()`. I think the ancestors iterator can be mapped to `(OperationId, Vec<GraphEdge<OperationId>>)` or `(Operation, Vec<GraphEdge<Operation>>)` which is compatible with `ReverseGraphIterator`.

It might make sense to reimplement `ReverseGraphIterator` to support `(T, Vec<GraphEdge<Id>>)` pair.